omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/SessionController1_8.java

```diff
@@ -61,13 +69,59 @@ public Object get(WebRequest request) {
 		if (authenticated) {
 			String repParam = request.getParameter(RestConstants.REQUEST_PROPERTY_FOR_REPRESENTATION);
 			Representation rep = (repParam != null) ? restService.getRepresentation(repParam) : Representation.DEFAULT;
-			session.add("user", ConversionUtil.convertToRepresentation(Context.getAuthenticatedUser(), rep));
 			session.add("locale", Context.getLocale());
+			session.add("currentProvider", ConversionUtil.convertToRepresentation(getCurrentProvider(), rep));
+			session.add("sessionLocation", ConversionUtil.convertToRepresentation(getLocation(request), rep));
+			session.add("user", ConversionUtil.convertToRepresentation(Context.getAuthenticatedUser(), rep));
 			session.add("allowedLocales", Context.getAdministrationService().getAllowedLocales());
 		}
 		return session;
 	}
-	
+
+	/**
+	 * Get current user selected location
+	 *
+	 * @param request web request
+	 * @return current user selected location
+	 */
+	private Location getLocation(WebRequest request){
+		Integer locationId = (Integer) request.getAttribute(LOCATION_SESSION_ATTRIBUTE, WebRequest.SCOPE_REQUEST);
+		Location sessionLocation = null;
+		if (locationId != null) {
+			sessionLocation = Context.getLocationService().getLocation(locationId);
+		}
```

Comment:
If the `locationId` is null, that means that the client has failed to provide the required parameter, right? I feel like that should cause a HTTP 400.

And if `Context.getLocationService().getLocation(locationId)` returns null, which it will if the user provides an invalid Location ID, this will simply set the sessionLocation to null and return 200. I think this is another case where we should return HTTP 400.